hibernate-core/src/main/antlr/org/hibernate/grammars/hql/HqlParser.g4

```diff
@@ -79,9 +79,9 @@ queryExpression
 	;
 
 queryGroup
-	: querySpec                                            										# QuerySpecQueryGroup
-	| LEFT_PAREN queryGroup RIGHT_PAREN															# NestedQueryGroup
-	| queryGroup queryOrder? (setOperator (querySpec | LEFT_PAREN queryGroup RIGHT_PAREN))+		# SetQueryGroup
+	: querySpec                                            												# QuerySpecQueryGroup
+	| LEFT_PAREN queryGroup queryOrder? RIGHT_PAREN														# NestedQueryGroup
+	| queryGroup queryOrder? (setOperator (querySpec | LEFT_PAREN queryGroup queryOrder? RIGHT_PAREN))+	# SetQueryGroup
```

Comment:
This rule looks like it could do with a refactoring, to remove the duplication of `queryGroup queryOrder?` and `LEFT_PAREN queryGroup queryOrder? RIGHT_PAREN`.

Split out a separate rule.

Reply:
Fixed